lib/matplotlib/axes/_axes.py

```diff
@@ -2622,8 +2622,10 @@ def bar_label(self, container, labels=None, *, fmt="%g", label_type="edge",
             A list of label texts, that should be displayed. If not given, the
             label texts will be the data values formatted with *fmt*.
 
-        fmt : str, default: '%g'
-            A format string for the label.
+        fmt : str or callable, default: '%g'
+            An unnamed %-style or {}-style format string for the label or a
+            function to call with the value as the first argument.
+            When *fmt* is a string, %-style takes precedence over {}-style.
```

Comment:
Let's make it very explicit when this is relevant.
```suggestion
            When *fmt* is a string and can be interpreted in both formats,
            %-style takes precedence over {}-style.
```